### DISCUSSION TITLE
Using the list on iPhone makes it prefer celluar over wifi

### DISCUSSION THREAD

**Discussion by rocketpear** [StevenBlack/hosts#2195] [Q&A]
So I noticed this weird behaviour. I have added this list to DNS filters in AdGuard on iPhone and it caused the iPhone to stop using WiFi even when showing it as connected in settings. It simply prefers LTE or 5G. Data works, and everything works, just over cellular instead of wifi. Other lists I use don't seem to have this effect.

I find this behaviour very strange and wondered what might be causing it.

**Comment by hagezi** [comment]
@Delltar 

How many rules do you use in AdGuard iOS? There is a limit for DNS of ~535,000 rules, if you use more, the behavior you describe will occur. VPN/Wifi crashes.

**Reply by rocketpear** [reply]
Ahh, that seems to be it! the Stephen Black list just brought my total number of rules over 550 000 (I just added a bunch of lists without checking overlaps tbh, so not really a best practice).
Thanks for clearing that up for me!
